public/app/core/utils/richHistory.ts

```diff
@@ -153,15 +152,27 @@ export const copyStringToClipboard = (string: string) => {
   document.body.removeChild(el);
 };
 
-export const createUrlFromRichHistory = (query: RichHistoryQuery) => {
-  const queries = query.queries.map(query => ({ expr: query }));
+export const createUrlFromRichHistory = (query: RichHistoryQuery, getQueryFromDisplayText?: (q: string) => {}) => {
```

Comment:
If `getQueryFromDisplayText` passed, use it to create url. Otherwise just parse queryString.